runtime/compiler/optimizer/J9EstimateCodeSize.cpp

```diff
@@ -1277,7 +1277,11 @@ TR_J9EstimateCodeSize::realEstimateCodeSize(TR_CallTarget *calltarget, TR_CallSt
       {
       bci.prepareToFindAndCreateCallsites(blocks, flags, callSites, &cfg, &newBCInfo, _recursionDepth, &callStack);
       bool iteratorWithState = isCalleeMethodHandleThunk && !mhInlineWithPeeking;
-      bci.findAndCreateCallsitesFromBytecodes(wasPeekingSuccessfull, iteratorWithState);
+      if (!bci.findAndCreateCallsitesFromBytecodes(wasPeekingSuccessfull, iteratorWithState))
+         {
+         heuristicTrace(tracer(), "*** Depth %d: ECS end for target %p signature %s. bci.findAndCreateCallsitesFromBytecode failed", _recursionDepth, calltarget, callerName);
+         return returnCleanup(7);
```

Comment:
why 7?

Reply:
I think the number is used to identify the location of error return starting from 0 and the largest existing number is 6.

Reply:
where are these defined? I'm not a fan of this kind of magic number...

Reply:
it's defined here https://github.com/eclipse/openj9/blob/ef5279e88d90827a8f4fbfe494e661539e7b74c0/runtime/compiler/optimizer/EstimateCodeSize.hpp#L81 and used here https://github.com/eclipse/openj9/blob/ef5279e88d90827a8f4fbfe494e661539e7b74c0/runtime/compiler/optimizer/InlinerTempForJ9.cpp#L3553-L3554
And inside https://github.com/eclipse/openj9/blob/ef5279e88d90827a8f4fbfe494e661539e7b74c0/runtime/compiler/optimizer/J9EstimateCodeSize.cpp#L1040 searching for `cleanup(` would show all the places using this API to bail. Probably a least an enum can make it looks cleaner.

Reply:
Ok, I think it would be good to create a cleanup item to at least convert these magic numbers to an enum as a beginner item since the current state of affairs is far from idea. Could we have a comment on the function where we use these magic numbers of what the return values are as a stop gap?

Reply:
added documentation for in the `returnCleanup` and opened an issue here https://github.com/eclipse/openj9/issues/8664